### DISCUSSION TITLE
2.3.11

### DISCUSSION THREAD

**Discussion by benhillis** [microsoft/WSL#11789] [General]
*  Add WSL Settings Feature
    *  Bridged networking mode is currently not selectable from the UI
*  Update Kernel to 6.6.36.3
    * Hundreds of new kernel modules added.
*  Update WSLg to 1.0.63
*  Update Microsoft.DXCore.Linux to version 10.0.26100.1-240331-1435.ge-release
*  Implement wsl --manage --move
*  From github feedback, increasing the Linux ephemeral port range (solves #11450)
*  Update mirrored mode loopback routing to support connections between Windows host and Linux Docker/Podman containers (solves #11136,#11468,#11758,#10926)
*  Propagate errors from bsdtar to stderr (solves #11577)
*  Improve error message when calling wsl.exe --exec (solves #11571)
*  Move kernel module mounting earlier in the boot process and adjust module mount behavior
*  Resolve issue with incorrect localization files for msrdc.exe
*  Disable virtio9p while an issue is being debugged (default is 9p with vsock transport)
*  Various fixes for VirtioFs
*  Improve socket protocol to catch errors early
*  Add error tracing statement for all cases where opening wslconfig fails
*  Change distro install to attempt store install then fall back to direct download

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/WSL/releases/tag/2.3.11'>2.3.11</a>.</em>

**Comment by AnonymousWP** [comment]
Does this release also solve these networking issues? https://github.com/microsoft/WSL/issues/11693

**Reply by craigloewen-msft** [reply]
Are you seeing networking issues today? Have you tried turning on mirrored mode networking?

**Reply by AnonymousWP** [reply]
> Are you seeing networking issues today? Have you tried turning on mirrored mode networking?

This fixed the issue for me: https://github.com/microsoft/WSL/issues/11693#issuecomment-2194230283, but it seems that different approaches work for people. For user A, approach Y works. For user B, approach Z works. Maybe going through the entire issue may help identifying what's going wrong. There are logs as well.

**Reply by craigloewen-msft** [reply]
Ok great! We're looking at ways to help improve this and are taking a look through those issues. Thanks for the feedback :)

**Comment by agxpro** [comment]
Windows Security flagging it as virus.

**Detected**: Trojan:Script/Wacatac.H!ml
**file**: C:\Users\abcxyz\Downloads\wsl.2.3.11.0.x64.msi
**webfile**: C:\Users\abcxyz\Downloads\wsl.2.3.11.0.x64.msi|about:internet|pid:14688,ProcessStart:133657257432458137

**Reply by davidfiala** [reply]
Dedicated issue at #11797

Interestingly, you report `Wacatac.H!ml` but others are seeing `Wacatac.B!ml`, a small variation.

**Reply by elsaco** [reply]
I did a `virustotal` scan of the `wsl.2.3.11.0.x64.msi` package and no warnings. See details a https://www.virustotal.com/gui/file/ac59215d7d723b226a6fbe9c85f88245e2d7355c7c8ec53fb8c0f1f4791d5d11

However, windows defender reports `Trojan:Script/Wacatac.B!ml` threat.

**Reply by agxpro** [reply]
> Dedicated issue at #11797
> 
> Interestingly, you report `Wacatac.H!ml` but others are seeing `Wacatac.B!ml`, a small variation.

Previously I'd "_Security Intelligence Update for Microsoft Defender Antivirus - KB2267602 (Version 1.415.113.0)_" installed when it flagged `Wacatac.H!ml` but now after updating to "_Security Intelligence Update for Microsoft Defender Antivirus - KB2267602 (Version 1.415.174.0)_" Windows Defender flagging `Wacatac.B!ml`.

**Comment by DoctorWho8** [comment]
Which is a false-flag. Please try another AV , I run NAV here which insisted that the file is safe.

**Reply by benhillis** [reply]
Agreed that doesn't look right to me either.

**Reply by DoctorWho8** [reply]
And also when I went ahead and did the install, the witless Windows Defender thing complained, it did not want me to run the installer. I did anyway, and everything works.

**Reply by newbenji** [reply]
Microsofts own defender also found it positive..

**Comment by heartacker** [comment]
https://github.com/microsoft/WSL/issues/11791

**Comment by dorssel** [comment]
> Hundreds of new kernel modules added

Until now, WSL was always built monolithic (i.e., stuff was either enabled with `CONFIG_XXX=y`, or disabled with `CONFIG_XXX=n`). Is it correct that modules are now actually supported (`CONFIG_XXX=m`)?

I am the author of `usbipd-win`, see issue https://github.com/dorssel/usbipd-win/issues/995. Is the USBIP support in WSL in future releases modularized?

On a side note, see also comment https://github.com/dorssel/usbipd-win/issues/995#issuecomment-2234856054: `CONFIG_USBIP_HOST=m` (or `=y`) is useless...

**Reply by kelsey-steele** [reply]
Modules are now supported! \o/ 

Important exception to this is that module support gets disabled (at least for now) if using a custom kernel. So, for anyone using custom kernels, please make sure all the modules you need are built in before building your kernel. 

These configs are set to continue into future WSL releases. There are some smaller planned config changes in the works, though not around the USBIP configs; you can expect those to remain as modules.

The new WSL2 kernel configs are designed to work with more than just WSL. So, this is why there are some select configs built-in or set as modules that don't quite make sense for WSL.

**Reply by VibroAxe** [reply]
This is great, but are the relevant kernel headers available/packaged to allow us to build modules such as usb-ip without cloning the entire kernel?

**Reply by benhillis** [reply]
Not yet, we're working on a story for how to have users provide their own modules. Expect to something here in the coming weeks.

**Comment by kristof-mattei** [comment]
I submitted a sample yesterday to MS Defender, and while it tested positive, a retest today shows nothing detected. Not sure if we need to update our definitions or whether some other mechanism will now allow the file.

**Reply by newbenji** [reply]
just retried with updated definities for ms defender. still positive

**Reply by kristof-mattei** [reply]
My definitions: `1.415.176.0`. I'm on Insiders Preview.

**Reply by elsaco** [reply]
With `Security Intelligence Update for Microsoft Defender Antivirus - KB2267602 (Version 1.415.176.0) - Current Channel (Broad)` it still triggers the threat.

**Comment by kirillkovalenko** [comment]
x64 MSI is not digitally signed.

**Reply by benhillis** [reply]
Interesting, I will investigate.

**Reply by kipters** [reply]
arm64 too

**Comment by g2flyer** [comment]
Seems for new versions user (`wsl`) and system (`wsl --system`) session get killed on each power-management event like sleep or hibernate while  `wsl --debug-shell` survives. Seems pretty deterministic; reverting to 2.2.4 makes wsl survive power-management events again.  I guess issue is new kernel?

**Reply by g2flyer** [reply]
Ah, seems there is already an issue for that: https://github.com/microsoft/WSL/issues/11806

**Reply by RonnBlack** [reply]
@g2flyer How did you revert.?
I saw one place that mentioned 

`wsl --update --rollback` 

but that doesn't seem to work

**Reply by RonnBlack** [reply]
The problem with Docker not starting may be fixed in 2.3.12 but there is a bigger problem that is not fixed that affects Kubernetes networking. 

I don't know why but it seems to affect `kube-dns` and `kube-proxy`. The symptoms are that pods can't find each other because the Kubernetes DNS lookups fail. The DNS lookup fails because the pod can't reach the DNS. Even if the DNS properly resolved the service, it can't reach the service via IP because the internal proxy isn't working correctly.

 I had to roll back to 2.2.4 to get it working again. For others, here is how I rolled back.

 1. Uninstall Windows Subsystem for Linux in either `Settings > Apps > Installed Apps` or `Settings > System > Optional Features > More Windows features` (Location depends on how you installed it). (Note: You don't need to un-register the distro)
 2. Locate the MSI file for the desired version from https://github.com/microsoft/releases.
 3. Download the MSI file and run the install

**Comment by DevKyleS** [comment]
Docker Desktop no longer starts after upgrading to this version, I had same issue as https://github.com/microsoft/WSL/issues/11813

**Comment by Sarisan** [comment]
WSL2 automatic shutdown ignores any running processes after this update.

**Reply by hez2010** [reply]
Can you try setting vmIdleTimeout to -1?

**Reply by Sarisan** [reply]
Didn't help

**Reply by astroboylrx** [reply]
confirm that `vmIdleTimeout=-1` didn't help

**Comment by jihoonkim2100** [comment]
Does this release solve [#11277](https://github.com/microsoft/WSL/issues/11277) issue? One user reported that it works - [link](https://github.com/microsoft/WSL/issues/11277#issuecomment-2240083713)

**Comment by vromanyu** [comment]
Is this getting pushed to stable once it passes all the tests? 

Thanks!
